frontend/src/diagram/DiagramWebSocketContainer.types.ts

```diff
@@ -59,8 +61,8 @@ export interface Palette {
 
 export interface Menu {
   canvasBounds: Bounds;
-  sourceElement: NodeDescription;
-  targetElement: NodeDescription;
+  sourceElement: Node;
+  targetElement: Node;
```

Comment:
I don't mind limiting the source and target diagram elements to Node but on the backend you may not need to look for potential edges in `GetConnectorToolsEventHandler#findDiagramElement`.